skopt/tests/test_forest_opt.py

```diff
@@ -40,33 +40,44 @@ def test_forest_minimize_api():
                          base_estimator=DecisionTreeClassifier())
 
 
-def check_minimize(minimizer, func, y_opt, dimensions, margin, n_calls):
-    # The result depends on random sampling so run the test several
-    # times and pass if majority of tests converge. Any single instance
-    # converging might just be luck.
-    success = 0
-    N = 3
-
-    for n in range(1, N + 1):
+def check_minimize(minimizer, func, y_opt, dimensions, margin,
+                        n_calls, n_random_starts=10, x0=None):
+    for n in range(3):
         r = minimizer(
-            func, dimensions, n_calls=n_calls, random_state=n)
-        if r.fun <= y_opt + margin:
-            success += 1
-
-    assert_less(N * 0.5, success)
-
+            func, dimensions, n_calls=n_calls, random_state=n,
+            n_random_starts=n_random_starts, x0=x0)
+        assert_less(r.fun, y_opt + margin)
 
 def test_tree_based_minimize():
     for name, minimizer in MINIMIZERS:
         yield (check_minimize, minimizer, bench1, 0.,
-               [(-2.0, 2.0)], 0.05, 25)
+               [(-2.0, 2.0)], 0.05, 25, 5)
+
+        # XXX: This benchmark contains 2 different functions
+        # when x < 0 and otherwise.
+        # Hence we supply points at the edge of the search
+        # space as an initial point to the minimizer.
```

Comment:
Not sure the reason is that the benchmark contains two functions. I think the more important part is getting samples on either side of the minimum. So I'd rephrase the comment to:

"Having samples on either side of the minimum is helpful for tree based methods. For this benchmark
the minimum is close to the edge and so we explicitly provide points at the boundaries of the search
space."

WDYT?